### PR TITLE
Fix Chartjs options support

### PR DESCRIPTION
**Pull request by tgalopin**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Tickets       | Fix https://github.com/symfony/ux/issues/11
| License       | MIT

Fix Chart.js options handling and add better tests to ensure no regression there.

### CODE REVIEW

File: src/Chartjs/Resources/assets/dist/controller.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ var _default = /*#__PURE__*/function (_Controller) {
     value: function connect() {
       var payload = JSON.parse(this.element.getAttribute('data-view'));
 
-      if (!payload.options.length) {
+      if (!Array.isArray(payload.options) && 0 === payload.options.length) {
```
Comment (Nyholm) [inline src/Chartjs/Resources/assets/dist/controller.js:58]: Hm.. If it is not an array AND length is zero.. 

Doesn't that mean that `payload.options = {}`?

Comment (tgalopin) [reply src/Chartjs/Resources/assets/dist/controller.js:58]: In JS, `[] !== {}`, which is the problem here, as `json_encode` creates `[]` when we would like `{}`

Comment (Nyholm) [reply src/Chartjs/Resources/assets/dist/controller.js:58]: > In JS, [] !== {},

Yes,

> as json_encode creates [] when we would like {}

But, that is not what I read... 

If `payload.options = []` then `!Array.isArray(payload.options)` will be false and you will never set `payload.options = {}`

Comment (Kocal) [reply src/Chartjs/Resources/assets/dist/controller.js:58]: What we can do is to always have a default value for `options`?

Comment (tgalopin) [reply src/Chartjs/Resources/assets/dist/controller.js:58]: Right! I have an extra `!`

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @tgalopin.

**Pull request merged**